pkg/kubelet/kubelet.go

```diff
@@ -2117,66 +2117,71 @@ func (kl *Kubelet) syncNetworkStatus() {
 	kl.networkConfigured = networkConfigured
 }
 
-// setNodeStatus fills in the Status fields of the given Node, overwriting
-// any fields that are currently set.
-func (kl *Kubelet) setNodeStatus(node *api.Node) error {
-	// Set addresses for the node.
-	if kl.cloud != nil {
-		instances, ok := kl.cloud.Instances()
-		if !ok {
-			return fmt.Errorf("failed to get instances from cloud provider")
-		}
-		// TODO(roberthbailey): Can we do this without having credentials to talk
-		// to the cloud provider?
-		// TODO(justinsb): We can if CurrentNodeName() was actually CurrentNode() and returned an interface
-		nodeAddresses, err := instances.NodeAddresses(kl.nodeName)
+func (kl *Kubelet) setUpdatedAddressesFromCloud(node *api.Node) {
+	instances, ok := kl.cloud.Instances()
+	if !ok {
+		glog.Errorf("Failed to get instances from cloud provider, so node addresses will be stale")
+		return
+	}
+	// TODO(roberthbailey): Can we do this without having credentials to talk to the cloud provider?
+	// TODO(justinsb): We can if CurrentNodeName() was actually CurrentNode() and returned an interface
+	nodeAddresses, err := instances.NodeAddresses(kl.nodeName)
+	if err != nil {
+		glog.Errorf("Failed to get addresses from cloud provider, so node addresses will be stale: %v", err)
+		return
+	}
+	node.Status.Addresses = nodeAddresses
+}
+
+func (kl *Kubelet) setUpdatedAddressesFromHostname(node *api.Node) {
+	addr := net.ParseIP(kl.hostname)
+	if addr == nil {
+		addrs, err := net.LookupIP(node.Name)
+
 		if err != nil {
-			return fmt.Errorf("failed to get node address from cloud provider: %v", err)
+			glog.Errorf("can't get ip address of node %s, so node addresses will be stale: %v", node.Name, err)
```

Comment:
Since this is a log statement, the first letter should be capitalized, e.g. "Can't get ..."